src/container.cpp

```diff
@@ -405,8 +403,8 @@ ReturnValue Container::queryRemove(const Thing& thing, uint32_t count, uint32_t
 	}
 
 	if (actor && getBoolean(ConfigManager::ONLY_INVITED_CAN_MOVE_HOUSE_ITEMS)) {
-		const Cylinder* const topParent = getTopParent();
-		if (const HouseTile* const houseTile = dynamic_cast<const HouseTile*>(topParent->getTile())) {
+		auto topParent = getTopParent();
```

Comment:
Could this be `const auto`
```suggestion
		const auto topParent = getTopParent();
```